drivers/flash/flash_simulator.c

```diff
@@ -436,14 +445,20 @@ static int flash_mock_init(const struct device *dev)
 }
 
 #else
-
+#if DT_NODE_HAS_PROP(DT_PARENT(SOC_NV_FLASH_NODE), memory_region)
+static int flash_mock_init(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+	return 0;
+}
+#else
 static int flash_mock_init(const struct device *dev)
 {
 	ARG_UNUSED(dev);
 	memset(mock_flash, FLASH_SIMULATOR_ERASE_VALUE, ARRAY_SIZE(mock_flash));
 	return 0;
 }
-
+#endif /* DT_NODE_HAS_PROP(DT_PARENT(SOC_NV_FLASH_NODE), skip_erase) */
```

Comment:
Something is missing in the `#if..#else..#endif` block, as the #if relies on existence of `memory-region` only, and here `skip-erase` property is mentioned.